# DP: Don't use any relative path names for the standard include paths.

--- a/src/gcc/incpath.c
+++ b/src/gcc/incpath.c
@@ -173,6 +173,14 @@ add_standard_paths (const char *sysroot,
 			str = reconcat (str, str, dir_separator_str,
 					imultiarch, NULL);
 		    }
+		  {
+		    char *rp = lrealpath (str);
+		    if (rp)
+		      {
+			free (str);
+			str = rp;
+		      }
+		  }
 		  add_path (str, INC_SYSTEM, p->cxx_aware, false);
 		}
 	    }
@@ -248,6 +256,14 @@ add_standard_paths (const char *sysroot,
 	      else
 		str = reconcat (str, str, dir_separator_str, imultiarch, NULL);
 	    }
+	  {
+	    char *rp = lrealpath (str);
+	    if (rp)
+	      {
+		free (str);
+		str = rp;
+	      }
+	  }
 
 	  add_path (str, INC_SYSTEM, p->cxx_aware, false);
 	}
